daemon/links/links_test.go

```diff
@@ -1,214 +1,101 @@
 package links // import "github.com/docker/docker/daemon/links"
 
 import (
-	"fmt"
-	"strconv"
-	"strings"
+	"sort"
 	"testing"
 
 	"github.com/docker/go-connections/nat"
+	"gotest.tools/v3/assert"
 )
 
-// Just to make life easier
-func newPortNoError(proto, port string) nat.Port {
-	p, _ := nat.NewPort(proto, port)
-	return p
-}
-
 func TestLinkNaming(t *testing.T) {
-	ports := make(nat.PortSet)
-	ports[newPortNoError("tcp", "6379")] = struct{}{}
-
-	link := NewLink("172.0.17.3", "172.0.17.2", "/db/docker-1", nil, ports)
-
-	rawEnv := link.ToEnv()
-	env := make(map[string]string, len(rawEnv))
-	for _, e := range rawEnv {
-		parts := strings.Split(e, "=")
-		if len(parts) != 2 {
-			t.FailNow()
-		}
-		env[parts[0]] = parts[1]
-	}
-
-	value, ok := env["DOCKER_1_PORT"]
-
-	if !ok {
-		t.Fatal("DOCKER_1_PORT not found in env")
-	}
-
-	if value != "tcp://172.0.17.2:6379" {
-		t.Fatalf("Expected 172.0.17.2:6379, got %s", env["DOCKER_1_PORT"])
-	}
+	link := NewLink("172.0.17.3", "172.0.17.2", "/db/docker-1", nil, nat.PortSet{
+		"6379/tcp": struct{}{},
+	})
+
+	expectedEnv := []string{
+		"DOCKER_1_NAME=/db/docker-1",
+		"DOCKER_1_PORT=tcp://172.0.17.2:6379",
+		"DOCKER_1_PORT_6379_TCP=tcp://172.0.17.2:6379",
+		"DOCKER_1_PORT_6379_TCP_ADDR=172.0.17.2",
+		"DOCKER_1_PORT_6379_TCP_PORT=6379",
+		"DOCKER_1_PORT_6379_TCP_PROTO=tcp",
+	}
+
+	actual := link.ToEnv()
+	sort.Strings(actual) // order of env-vars is not relevant
+	assert.DeepEqual(t, expectedEnv, actual)
 }
 
 func TestLinkNew(t *testing.T) {
-	ports := make(nat.PortSet)
-	ports[newPortNoError("tcp", "6379")] = struct{}{}
-
-	link := NewLink("172.0.17.3", "172.0.17.2", "/db/docker", nil, ports)
+	link := NewLink("172.0.17.3", "172.0.17.2", "/db/docker", nil, nat.PortSet{
+		"6379/tcp": struct{}{},
+	})
 
-	if link.Name != "/db/docker" {
-		t.Fail()
-	}
-	if link.ParentIP != "172.0.17.3" {
-		t.Fail()
-	}
-	if link.ChildIP != "172.0.17.2" {
-		t.Fail()
-	}
-	for _, p := range link.Ports {
-		if p != newPortNoError("tcp", "6379") {
-			t.Fail()
-		}
+	expected := &Link{
+		Name:     "/db/docker",
+		ParentIP: "172.0.17.3",
+		ChildIP:  "172.0.17.2",
+		Ports:    []nat.Port{"6379/tcp"},
 	}
+
+	assert.DeepEqual(t, expected, link)
```

Comment:
FWIW; this is roughly how I arrived in this package; the `Link` type is not used anywhere, and it's only used as an intermediate struct to produce the env-vars to set.

I may be rewriting some code so that the scope of the package is reduced to only generating env-vars and internalise the `Link` type.